File: Part1_강의자료/Ch05_정렬/문제별코드/1302_베스트셀러/1302.cpp
#include <stdio.h>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

int main()
{
    int N;
    vector<string> titles;
    char title[51];

    scanf("%d", &N);
    for (int i = 0; i < N; i++)
    {
        scanf("%s", title);
        titles.push_back(title);
    }

    sort(titles.begin(), titles.end());

    string max_title = titles[0];
    int max_cnt = 1;
    int cur_cnt = 1;
    for (int i = 1; i < N; i++)
    {
        if (titles[i - 1] != titles[i])
            cur_cnt = 0;
        cur_cnt++;
        if (max_cnt < cur_cnt || (max_cnt == cur_cnt && titles[i] < max_title))
        {
            max_cnt = cur_cnt;
            max_title = titles[i];
        }
    }

    printf("%s\n", max_title.c_str());
    return 0;
}

File: Part1_강의자료/Ch05_정렬/문제별코드/1302_베스트셀러/1302.py
N = int(input())
titles = []

for i in range(N):
    title = input().strip()
    titles.append(title)

titles.sort()

max_title = titles[0]
max_cnt = 1
cur_cnt = 1
for i in range(1, N):
    if titles[i - 1] != titles[i]:
        cur_cnt = 0
    cur_cnt += 1
    if max_cnt < cur_cnt or (max_cnt == cur_cnt and titles[i] < max_title):
        max_cnt, max_title = cur_cnt, titles[i]

print(max_title)
